packages/client-purpose-updater/src/clientPurposeUpdaterConsumerServiceV2.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  CorrelationId,
+  generateId,
+  PurposeEventEnvelopeV2,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import {
+  getInteropHeaders,
+  RefreshableInteropToken,
+} from "pagopa-interop-commons";
+import { getInteropBeClients } from "./clients/clientsProvider.js";
+
+const { authorizationClient } = getInteropBeClients();
+
+export async function handleMessageV2(
+  message: PurposeEventEnvelopeV2,
+  refreshableToken: RefreshableInteropToken
+): Promise<void> {
+  await match(message)
+    .with({ type: "PurposeArchived" }, async (purposeMsg) => {
+      const token = (await refreshableToken.get()).serialized;
+      const correlationId = generateId<CorrelationId>();
+      const headers = getInteropHeaders({ token, correlationId });
```

Comment:
```suggestion
      const headers = getInteropHeaders({
        token,
        correlationId: generateId<CorrelationId>(),
      });
```
Since `correlationId` is not reused you can generate it directly there